### ISSUE TITLE
TryParse Throws Exceptions

### ISSUE THREAD

**Issue by cvergaray** [Tronald/CoordinateSharp#146]
**Describe the bug**
The Coordinate.TryParse() method will throw an exception if the input is null or empty.

**To Reproduce**
Here is an NUnit3 test case that demonstrates the issue:
```
/// <summary>
/// Ensures that the <see cref="CoordinateSharp.Coordinate.TryParse" /> method does
/// not throw an exception when the provided string is null or empty.
/// </summary>
[Test]
public void TryParseDoesNotThrowWhenNullOrEmpty()
{
    Assert.Multiple(() =>
    {
        Assert.Throws<IndexOutOfRangeException>(() => { _ = CoordinateSharp.Coordinate.Parse(" "); });
        Assert.DoesNotThrow(() => { CoordinateSharp.Coordinate.TryParse(" ", out var _); });

        Assert.Throws<NullReferenceException>(() => { _ = CoordinateSharp.Coordinate.Parse(null); });
        Assert.DoesNotThrow(() => { CoordinateSharp.Coordinate.TryParse(null, out var _); });

    });
}
```

**Expected behavior**
I expect that the TryParse method would never throw an exception but rather always return a boolean indicating success/failure.

**Environment:**
- .NET version: Standard 2.0
- CoordinateSharp version: 2.6.2.1
- Application type: Xamarin Mobile Application (iOS, Android, & UWP) for Military response operations
- Is mutlithreading being used: Yes

**Additional context**
I know it's another small one, but thanks for considering! We have the try parse code connected to the user-supplied value in a text box. We use the try parse method to determine if the provided value is actually a valid coordinate. We could wrap everything in a try/catch, but that's what I expect the TryParse would do for us.

**Comment by Tronald** [comment]
Agreed, TryParse should definitely be able to handle null or empty. I appreciate everything you find as it helps to improve the library.

We’ll get that patched up for you with the next release.

**Comment by Tronald** [comment]
Upon further investigation it appears this bug was created with the addition of UPS/MGRS Polar systems being added to the parsers in v2.4.3.1. Those parsers did not include logic for empty string and no testing was added to catch the situation. We're going to add a null/empty check at the beginning of the parse operation to handle all future scenarios.

Side note, we're also starting the move from the console testing project to a unit test one in an effort to improve testing. Thanks again.

**Comment by Tronald** [comment]
Fixed with [v2.7.1.1](https://github.com/Tronald/CoordinateSharp/releases/tag/v2.7.1.1) release on 12-JUL-2020.
